src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.cs

```diff
@@ -0,0 +1,128 @@
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (!RequiredSymbols.TryGetSymbols(context.Compilation, out var symbols))
+                    return;
+
+                var selector = UseStringEqualsOverStringCompareFixer.GetOperationSelector(symbols);
+
+                context.RegisterOperationAction(context =>
+                {
+                    if (selector(context.Operation))
+                    {
+                        var diagnostic = context.Operation.CreateDiagnostic(Rule);
+                        context.ReportDiagnostic(diagnostic);
+                    }
+                }, OperationKind.Binary);
+            });
+        }
+
+        internal sealed class RequiredSymbols
+        {
+            //  Named-constructor 'TryGetSymbols' inits all properties or doesn't construct an instance.
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+            private RequiredSymbols() { }
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+
+            public static bool TryGetSymbols(Compilation compilation, [NotNullWhen(true)] out RequiredSymbols? symbols)
+            {
+                symbols = default;
+
+                var stringType = compilation.GetSpecialType(SpecialType.System_String);
+                var boolType = compilation.GetSpecialType(SpecialType.System_Boolean);
+
+                if (stringType is null || boolType is null)
+                    return false;
+
+                if (!compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparison, out var stringComparisonType))
+                    return false;
+
+                var compareMethods = stringType.GetMembers(nameof(string.Compare))
+                    .OfType<IMethodSymbol>()
+                    .Where(x => x.IsStatic);
+                var compareStringString = compareMethods.GetFirstOrDefaultMemberWithParameterTypes(stringType, stringType);
+                var compareStringStringBool = compareMethods.GetFirstOrDefaultMemberWithParameterTypes(stringType, stringType, boolType);
+                var compareStringStringStringComparison = compareMethods.GetFirstOrDefaultMemberWithParameterTypes(stringType, stringType, stringComparisonType);
+
+                if (compareStringString is null ||
+                    compareStringStringBool is null ||
+                    compareStringStringStringComparison is null)
+                {
+                    return false;
+                }
```

Comment:
This would be a false negative if one of the overloads exists but not the others.
However, it doesn't worth fixing that IMO since all these overloads are expected to almost always exist.

Reply:
@Youssef1313 Good catch. I'm having to basically rewrite the analyzer due to architecture flaws (I didn't realize analyzers couldn't reference anything in the code fix), but I will make sure to rewrite it so it is tolerant of some overloads being absent.